src/figuresClasses.ts

```diff
@@ -2,10 +1,79 @@
+export interface Figure {
+  shape: string;
+  color: string;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: string = 'triangle';
+
+  constructor(
+    public color: string,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const sides = [a, b, c];
+    const maxSide = Math.max(a, b, c);
+    const leftSides = sides.filter((side) => side !== maxSide);
+    const sumLeftSides = leftSides.reduce((acc, side) => acc + side);
+
+    if (a <= 0 || b <= 0 || c <= 0 || sumLeftSides <= maxSide) {
```

Comment:
The error message should be more descriptive. Consider specifying which condition failed or providing a message like 'Invalid triangle sides'.